### DISCUSSION TITLE
Enhancing Error Troubleshooting by Adding Traceback to Exception Logs

### DISCUSSION THREAD

**Discussion by pythrick** [Lancetnik/Propan#110] [Ideas]
Hello,
I've been working with the code and noticed an opportunity to improve our error logging and debugging experience.
### Current Code Handling:
Below is how our current code handles and logs exceptions:
https://github.com/Lancetnik/Propan/blob/fd243dabd0cec5a46fbe511e4321c6df1a618dd3/propan/brokers/_model/broker_usecase.py#L614-L642

In its current state, the code logs exceptions but doesn't include a traceback or the log_context that could potentially be very useful for debugging.
### My Proposal:
To enhance our error logging, I suggest adding the traceback to exception logs, and to provide extra context, include log_context. Here's a potential way of doing it:

```python
# ...
                try:
                    r = await func(message)
                except SkipMessage as e:
                    self._log("Skipped", extra=log_context)
                    raise e
                except Exception as e:
                    self._log(repr(e), logging.ERROR, extra=log_context, exc_info=e)
                    raise e
# ...
    def _log(
        self,
        message: str,
        log_level: Optional[int] = None,
        extra: Optional[AnyDict] = None,
        exc_info: BaseException | bool | None = None,
    ) -> None:
        if self.logger is not None:
            self.logger.log(
                level=(log_level or self.log_level), msg=message, extra=extra, exc_info=exc_info
            )
```
By incorporating these, we could have better insights into errors which could streamline our debugging process.

### Looking for Suggestions:

The part I am pondering over is how to allow users to opt in or out of this additional logging behavior, maybe using an environment variable.

I would love to hear your thoughts and any potential solutions for this part. Thank you in advance for your input.

**Comment by Lancetnik** [comment]
It is just a decision I use in a new major draft😅
Can u make a PR to implement it here too?

**Comment by Lancetnik** [comment]
So, you can find it in 0.1.5.14

**Reply by pythrick** [reply]
Thanks!
